flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -913,6 +920,18 @@ public LinkedHashSet<Class<?>> getRegisteredPojoTypes() {
         return registeredPojoTypes;
     }
 
+    /**
+     * Get if the auto type registration is disabled.
+     *
+     * @return if the auto type registration is disabled.
+     * @deprecated All Flink DataSet APIs are deprecated since Flink 1.18 and will be removed in a
+     *     future Flink major version. You can still build your application in DataSet, but you
+     *     should move to either the DataStream and/or Table API.
+     * @see <a href="https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=158866741">
+     *     FLIP-131: Consolidate the user-facing Dataflow SDKs/APIs (and deprecate the DataSet
+     *     API</a>
+     */
+    @Deprecated
```

Comment:
We should also deprecated `disableAutoTypeRegistration`, and `AUTO_TYPE_REGISTRATION`.

Reply:
Both `disableAutoTypeRegistration ` and `AUTO_TYPE_REGISTRATION ` are deprecated.